src/views/annual-report/annual-report.jsx

```diff
@@ -0,0 +1,1279 @@
+    leadership: <FormattedMessage id="annualReport.subnavLeadership" />,
+    donate: <FormattedMessage id="annualReport.subnavDonate" />
+};
+
+// Constants used for world map data processing/formatting for use with Plotly
+const countryKeys = Object.keys(CountryUsage);
+const countryNames = countryKeys.map(key => CountryUsage[key].display);
+const countryData = countryKeys.map(key => CountryUsage[key].count);
+const colorIndex = countryKeys.map(key => CountryUsage[key]['log count']);
+
+class AnnualReport extends React.Component {
+    constructor (props) {
+        super(props);
+
+        // Storage for each of the section refs when we need to refer
+        // to them in the scroll handling code
+        // These will be stored with a short lowercase key representing
+        // the specific section (e.g. 'mission')
+        this.sectionRefs = {};
+
+        this.subnavRef = null;
+
+        this.state = {
+            currentlyVisible: SECTIONS.message, // The currently visible section
+            dropdownVisible: false
+        };
+
+        bindAll(this, [
+            'scrollTo',
+            'setRef',
+            'setSubnavRef',
+            'handleSubnavItemClick',
+            'getDimensionsOfSection',
+            'handleScroll',
+            'handleDropDownClick'
+        ]);
+    }
+
+    componentDidMount () {
+        window.addEventListener('scroll', this.handleScroll);
+    }
+
+    componentWillUnMount () {
+        window.removeEventListener('scroll', this.handleScroll);
+    }
+
+    // A generic handler for a subnav item that takes the name of the
+    // section to scroll to (all lowercase)
+    handleSubnavItemClick (sectionName) {
+        // Return a button click handler that will close the dropdown if open
+        // and scrolls to the correct section
+        return () => {
+            this.setState({dropdownVisible: false});
+            this.scrollTo(this.sectionRefs[sectionName]);
+        };
+    }
+
+    scrollTo (element) {
+        if (element) {
+            element.scrollIntoView({
+                behavior: 'smooth',
+                block: 'start'
+            });
+        }
+    }
+
+    // Generically create a ref for the given section, stored in
+    // this.sectionRefs
+    setRef (sectionName) {
+        return ref => (this.sectionRefs[sectionName] = ref);
+    }
+
+    setSubnavRef (ref) {
+        this.subnavRef = ref;
+    }
+
+    // Calculate the dimensions of a given section for use in figuring out
+    // which section is currently visible
+    getDimensionsOfSection (sectionRef) {
+        const {height} = sectionRef.getBoundingClientRect();
+        const offsetTop = sectionRef.offsetTop;
+        const offsetBottom = offsetTop + height;
+
+        return {
+            height,
+            offsetTop,
+            offsetBottom
+        };
+    }
+
+    // While scrolling, update the subnav to reflect the currently visible section
+    handleScroll () {
```

Comment:
Everything involving the scrolling and the subnav bar are probably the most complex part of the code, especially this function.